### PR TITLE
adding dynamic sitemap to read whole pages path and title

### PR DESCRIPTION
**Pull request by ibexad**

### CODE REVIEW

**Comment by MikeMcC399** [comment]
@ibexad 
You should probably exclude 
```
        src/data/sitemaplist.json
        src/data/sitemaplist_de.json
```
from being committed to the repository, through `/src/data/.gitignore`, since the files are regenerated by building the web.

**Comment by MikeMcC399** [comment]
The generated sitemap is not very readable as it has only one level and shows no structure.

**Comment by MikeMcC399** [comment]
Since there are new packages added to `package.json`, I would also expect an updated `package-lock.json` to be part of this PR. 

I cannot see any changed `package-lock.json` in the PR.

**Comment by MikeMcC399** [comment]
The generated sitemap makes the inconsistent use of hyphen / en dash in `page-title` visible:

image

For instance
https://github.com/corona-warn-app/cwa-website/blob/1ecd5f70d99cc3f0b65212a9b16595c467aed21d/src/pages/en/analysis/index.html#L3
uses `-`  or `U+002D` / Hyphen-Minus

whereas
https://github.com/corona-warn-app/cwa-website/blob/1ecd5f70d99cc3f0b65212a9b16595c467aed21d/src/pages/en/screenshots/index.html#L3
uses `–` or `U+2013` / En Dash

**Comment by MikeMcC399** [comment]
@ibexad 

You need to be using node 16 (see https://github.com/corona-warn-app/cwa-website/blob/master/README.md#requirements) otherwise `package-lock.json` is generated using an old version, due to the older version of `npm`.

File: src/data/sitemaplist.json [hunk 1]
```diff
@@ -0,0 +1,110 @@
+[
```
Comment (MikeMcC399) [inline src/data/sitemaplist.json:1]: This file should be removed from the PR.

File: src/data/sitemaplist_de.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
+[
```
Comment (MikeMcC399) [inline src/data/sitemaplist_de.json:1]: Also this file should be removed from the PR.

**Comment by MikeMcC399** [comment]
@ibexad

It seems that this PR is still a work-in-progress. Do you want any more comments on it at this time? Perhaps if you know that you still want to make changes to it, you could convert it to draft status, so it is clear that it is not ready to be merged?

In any case I noticed various points:

1. Individual pages from `/en/science/` are not included in the sitemap. They should probably be listed.
2. The redirects, such as `/en/faq/(17).html`, are included in the sitemap although they do not point to unique content. They should be removed from the sitemap.
3. `/en/news/index.html` is a redirect to `/en/blog` so should probably be removed.
4. The screenshot archive pages, such as `/en/screenshots-archive/2-10.html` all have identical page titles, which do not include the version.
5. The sorting of the screenshot pages is in ascending order (2-10, 2-11, 2-12, 2-13, 2-14, 2-15, 2-16, 2-17, 2-2, 2-3, 2-4, 2-5, 2-6, 2-7, 2-8, 2-9) but in two groups with two-digit minor versions first, followed by one-digit minor versions. The one-digit minor versions should appear first in the list to make the sorting order consistently ascending.

Where I referred to `/en` above, the same applies to `/de`.

**Comment by ibexad** [comment]
> @ibexad
> 
> It seems that this PR is still a work-in-progress. Do you want any more comments on it at this time? Perhaps if you know that you still want to make changes to it, you could convert it to draft status, so it is clear that it is not ready to be merged?
> 
> In any case I noticed various points:
> 
> 1. Individual pages from `/en/science/` are not included in the sitemap. They should probably be listed.
> 2. The redirects, such as `/en/faq/(17).html`, are included in the sitemap although they do not point to unique content. They should be removed from the sitemap.
> 3. `/en/news/index.html` is a redirect to `/en/blog` so should probably be removed.
> 4. The screenshot archive pages, such as `/en/screenshots-archive/2-10.html` all have identical page titles, which do not include the version.
> 5. The sorting of the screenshot pages is in ascending order (2-10, 2-11, 2-12, 2-13, 2-14, 2-15, 2-16, 2-17, 2-2, 2-3, 2-4, 2-5, 2-6, 2-7, 2-8, 2-9) but in two groups with two-digit minor versions first, followed by one-digit minor versions. The one-digit minor versions should appear first in the list to make the sorting order consistently ascending.
> 
> Where I referred to `/en` above, the same applies to `/de`.

Hello and thank you for your comments. 
I resolved all of them except item number 5, sorting of screenshot pages. I couldn't sort them when they are generated by gulp libraries and also handlebar has not had an option to sort the object on the page using js. Could you please take a deeper look and let me know if you find a way to do that. Thank you so much.

**Comment by MikeMcC399** [comment]
@ibexad 
Congratulations on your progress! It is good that you added version numbers to the screenshot titles. I am sorry that I don't have a solution for the screenshot version sorting. I think it is acceptable to leave it like it is now though.

I will add some other comments separately, so it is easier to respond to them.

File: src/data/sitemap_de.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        "headline": {
+            "title": "Sitemap"
+        },
+        "content": {
+            "category": {
+                "homepage" : "Startseite",
+                "dashboard" : "Dashboard",
+                "faq" : "FAQ",
+                "blog" : "Blog",
+                "news": "Nachrichten",
```
Comment (MikeMcC399) [inline src/data/sitemap_de.json:16]: The line for "science" is missing here, so there is no header for "Science" shown on `/de/sitemap/`.

**Comment by MikeMcC399** [comment]
`/de/terms-of-use/` is missing from the German `/de/sitemap/`

The file https://github.com/cuisines/cwa-website/blob/sitemap/src/pages/de/terms-of-use/index.html needs to be modified, like https://github.com/cuisines/cwa-website/blob/2c6fabc3a8fd1da6cf9e6f7bd86c81e4f49aca68/src/pages/en/terms-of-use/index.html#L6.

`page-category: About`

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
There are two problems on the German sitemap:

- header for Science missing
- `terms-of-use` / Nutzungsbedingungen missing

**Comment by ibexad** [comment]
> 

I did not get it, could you please explain what is the issue with the german sitemap?

> There are two problems on the German sitemap:
> 
> * header for Science missing
> * `terms-of-use` / Nutzungsbedingungen missing

File: src/data/sitemap_de.json (hunk 3 above)
Comment (MikeMcC399) [reply src/data/sitemap_de.json:17]: You can mark this comment as resolved now.
Thank you!

**Comment by MikeMcC399** [comment]
@ibexad 

Thank you for fixing the issues with the German sitemap. I guess your previous comment "I did not get it, could you please explain what is the issue with the german sitemap?" is now out of date.

Let me know if there is still anything unclear.

**Comment by MikeMcC399** [comment]
@ibexad 

- Will add the new pages https://www.coronawarn.app/en/accessibility/ and https://www.coronawarn.app/de/accessibility/ from PR https://github.com/corona-warn-app/cwa-website/pull/2508 to the sitemap?

**Review by svengabr — APPROVED** [review]

**Comment by MikeMcC399** [comment]
@svengabr 

After this merge, critical security vulnerabilities are reported.
`npm audit`
`43 vulnerabilities (1 low, 18 moderate, 19 high, 5 critical)`

The sitemap itself shows only a top level

image

I'm not sure where "News" came from and the FAQ section is not listed.

**Pull request merged**